File: frontend/src/types/movie.ts
export interface Movie {
  id: number;      
  title: string;
  year: number;
  director: string;
  image_url?: string;
  rating: number;  
}

export interface MovieFormData {
  title: string;
  year: number;
  director: string;
  rating: number;    
}

export interface ApiMovieResponse {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt: null | string;
  title: string;
  year: number;
  director: string;
  rating: number;
  image_url?: string;
}

File: backend/models/movie.go
package models

import "gorm.io/gorm"

type Movie struct {
	gorm.Model
	Title    string  `json:"title" binding:"required"`
	Director string  `json:"director" binding:"required"`
	Year     int16   `json:"year" binding:"required"`
	Rating   float32 `json:"rating" binding:"required"`
	ImageURL string  `json:"image_url"`
}
